### PR TITLE
Closes #620 | Add Dataloader thai-romanization

### LINKED ISSUE #620: Create dataset loader for thai-romanization
Dataloader name: `thai_romanization/thai_romanization.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?thai_romanization

|     Dataset| thai_romanization |
|-------------|---|
| Description | The Thai Romanization dataset contains 648,241 Thai words that were transliterated into English, making Thai pronounciation easier for non-native Thai speakers. This is a valuable dataset for Thai language learners and researchers working on Thai language processing task. Each word in the Thai Romanization dataset is paired with its English phonetic representation, enabling accurate pronunciation guidance. This facilitates the learning and practice of Thai pronunciation for individuals who may not be familiar with the Thai script. The dataset aids in improving the accessibility and usability of Thai language resources, supporting applications such as speech recognition, text-to-speech synthesis, and machine translation. It enables the development of Thai language tools that can benefit Thai learners, tourists, and those interested in Thai culture and language. |
| Subsets | - |
| Languages | tha, eng |
| Tasks | Word lists |
| License | Creative Commons Attribution Share Alike 3.0 (cc-by-sa-3.0) |
| Homepage | https://www.kaggle.com/datasets/wannaphong/thai-romanization/data |
| HF URL | - |
| Paper URL | - |

### PR DESCRIPTION
**Pull request by muhammadravi251001**
**Title**: Add Dataloader thai-romanization

**First line PR Message**: Closes https://github.com/SEACrowd/seacrowd-datahub/issues/620

**Note**: 
- There are no citations available for this dataset.
- This task is more like `transliteration` task, not a `word list` task --> transliterate Thai language to Roman (common alphabet used); that's why this dataset called as "romanization".

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/{my_dataset}/{my_dataset}.py` (please use only lowercase and underscore for dataset folder naming, as mentioned in dataset issue) and its `__init__.py` within `{my_dataset}` folder.
- [x] Provide values for the  `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_LOCAL`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py` or `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py --subset_id {subset_name_without_source_or_seacrowd_suffix}`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/thai_romanization/thai_romanization.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  author    = {},
+  title     = {},
+  journal   = {},
+  volume    = {},
+  year      = {},
+  url       = {},
+  doi       = {},
+  biburl    = {},
+  bibsource = {}
+}
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/thai_romanization/thai_romanization.py:56]: Maybe make this an empty string or `None`?

File: seacrowd/sea_datasets/thai_romanization/thai_romanization.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    Each word in the Thai Romanization dataset is paired with \
+    its English phonetic representation, enabling accurate \
+    pronunciation guidance. This facilitates the learning and \
+    practice of Thai pronunciation for individuals who may not \
+    be familiar with the Thai script. The dataset aids in improving \
+    the accessibility and usability of Thai language resources, \
+    supporting applications such as speech recognition, text-to-speech \
+    synthesis, and machine translation. It enables the development of \
+    Thai language tools that can benefit Thai learners, tourists, \
+    and those interested in Thai culture and language.
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/thai_romanization/thai_romanization.py:112]: I think it's fine to have the long description in the `_DESCRIPTION` and the docstring here should just describe the dataloader

```suggestion
    Thai Romanization dataloader from Kaggle (Phong et al., 2018)
```

**Review by ljvmiranda921 — APPROVED** [review]
LGTM! Just have a few minor comments but good to go 👍

File: seacrowd/sea_datasets/thai_romanization/thai_romanization.py (hunk 2 above)
Comment (muhammadravi251001) [reply seacrowd/sea_datasets/thai_romanization/thai_romanization.py]: Solved on https://github.com/SEACrowd/seacrowd-datahub/pull/645/commits/e1500c082185807f49cbf01bcdde7eab99578869 commit, Sir.

File: seacrowd/sea_datasets/thai_romanization/thai_romanization.py (hunk 1 above)
Comment (muhammadravi251001) [reply seacrowd/sea_datasets/thai_romanization/thai_romanization.py]: Solved on https://github.com/SEACrowd/seacrowd-datahub/pull/645/commits/805a0fb0c92e7284e38d99571bc1c5fdefea9451 commit, Sir.

**Review by MJonibek — APPROVED** [review]
@muhammadravi251001 Great work, thank you. LGTM

**Pull request merged**